tests/Integration/Behaviour/Features/Context/ShopFeatureContext.php

```diff
@@ -49,19 +53,138 @@ public function shopWithNameExists(string $reference, string $shopName)
         SharedStorage::getStorage()->set($reference, new Shop($shopId));
     }
 
+    /**
+     * @Given /I add a shop group "(.+)" with name "(.+)"$/
+     *
+     * @param string $reference
+     * @param string $groupName
+     */
+    public function addShopGroup(string $reference, string $groupName): void
+    {
+        $shopGroup = new ShopGroup();
+        $shopGroup->name = $groupName;
+        $shopGroup->active = true;
+        if (!$shopGroup->add()) {
+            throw new RuntimeException(sprintf('Could not create shop group: %s', Db::getInstance()->getMsgError()));
+        }
+
+        SharedStorage::getStorage()->set($reference, $shopGroup);
+    }
+
+    /**
+     * @Given /I add a shop "(.+)" with name "(.+)" for the group "(.+)"$/
+     *
+     * @param string $reference
+     * @param string $shopName
+     * @param string $shopGroupName
+     */
+    public function addShop(string $reference, string $shopName, string $shopGroupName): void
+    {
+        $shop = new Shop();
+        $shop->active = true;
+        $shop->id_shop_group = ShopGroup::getIdByName($shopGroupName);
+        $shop->id_category = 2;
```

Comment:
Could we explain this magic number? (either a constant or a comment)

Reply:
Fixed

Reply:
I believe you can use `Configuration::get('PS_HOME_CATEGORY')` to be a bit more dynamic